### ISSUE TITLE
[no-extra-parens] TypeError: rules.ForInStatement, ForOfStatement is not a function

### ISSUE THREAD

**Issue by 6XGate** [typescript-eslint/typescript-eslint#3452]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

I'm trying to create a pull request for Standard TS and this error was trigger linting their tests:

- [Repo branch](https://github.com/6XGate/eslint-config-standard-with-typescript/tree/feature/propagate-all-extension-rules)
- ```
  > eslint-config-standard-with-typescript@20.0.0 lint
  > eslint --ext .js,.ts .
  
  
  Oops! Something went wrong! :(
  
  ESLint: 7.27.0
  
  TypeError: rules.ForInStatement, ForOfStatement is not a function
  Occurred while linting /home/matthew/Projects/eslint-config-standard-with-typescript/src/index.test.ts:173
      at ForInStatement, ForOfStatement (/home/matthew/Projects/eslint-config-standard-with-typescript/node_modules/@typescript-eslint/eslint-plugin/dist/rules/no-extra-parens.js:119:63)
      at /home/matthew/Projects/eslint-config-standard-with-typescript/node_modules/eslint/lib/linter/safe-emitter.js:45:58
      at Array.forEach (<anonymous>)
      at Object.emit (/home/matthew/Projects/eslint-config-standard-with-typescript/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
      at NodeEventGenerator.applySelector (/home/matthew/Projects/eslint-config-standard-with-typescript/node_modules/eslint/lib/linter/node-event-generator.js:293:26)
      at NodeEventGenerator.applySelectors (/home/matthew/Projects/eslint-config-standard-with-typescript/node_modules/eslint/lib/linter/node-event-generator.js:322:22)
      at NodeEventGenerator.enterNode (/home/matthew/Projects/eslint-config-standard-with-typescript/node_modules/eslint/lib/linter/node-event-generator.js:336:14)
      at CodePathAnalyzer.enterNode (/home/matthew/Projects/eslint-config-standard-with-typescript/node_modules/eslint/lib/linter/code-path-analysis/code-path-analyzer.js:711:23)
      at /home/matthew/Projects/eslint-config-standard-with-typescript/node_modules/eslint/lib/linter/linter.js:954:32
      at Array.forEach (<anonymous>)
  ```
<!--
Include a ***minimal*** reproduction case.
The more irrelevant code/config you give, the harder it is for us to investigate.

Please consider creating an isolated reproduction repo to make it easy for the volunteer maintainers debug your issue.
-->

All of this is provided in the repo as best I can figure.

<!--
Also include your tsconfig, if you're using type-aware linting
-->

**Expected Result**

<!--
What did you expect to happen?
Please be specific here - list the exact lines and messages you expect.
-->

No error should occur.

**Actual Result**

<!--
What actually happened?
Please be specific here - list the exact lines and messages that caused errors
-->

```
  > eslint-config-standard-with-typescript@20.0.0 lint
  > eslint --ext .js,.ts .

  Oops! Something went wrong! :(

  ESLint: 7.27.0

  TypeError: rules.ForInStatement, ForOfStatement is not a function
  Occurred while linting /home/matthew/Projects/eslint-config-standard-with-typescript/src/index.test.ts:173
      at ForInStatement, ForOfStatement (/home/matthew/Projects/eslint-config-standard-with-typescript/node_modules/@typescript-eslint/eslint-plugin/dist/rules/no-extra-parens.js:119:63)
      at /home/matthew/Projects/eslint-config-standard-with-typescript/node_modules/eslint/lib/linter/safe-emitter.js:45:58
      at Array.forEach (<anonymous>)
      at Object.emit (/home/matthew/Projects/eslint-config-standard-with-typescript/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
      at NodeEventGenerator.applySelector (/home/matthew/Projects/eslint-config-standard-with-typescript/node_modules/eslint/lib/linter/node-event-generator.js:293:26)
      at NodeEventGenerator.applySelectors (/home/matthew/Projects/eslint-config-standard-with-typescript/node_modules/eslint/lib/linter/node-event-generator.js:322:22)
      at NodeEventGenerator.enterNode (/home/matthew/Projects/eslint-config-standard-with-typescript/node_modules/eslint/lib/linter/node-event-generator.js:336:14)
      at CodePathAnalyzer.enterNode (/home/matthew/Projects/eslint-config-standard-with-typescript/node_modules/eslint/lib/linter/code-path-analysis/code-path-analyzer.js:711:23)
      at /home/matthew/Projects/eslint-config-standard-with-typescript/node_modules/eslint/lib/linter/linter.js:954:32
      at Array.forEach (<anonymous>)
```

**Additional Info**

<!--
Did eslint throw an exception?

Please run your lint again with the --debug flag, and dump the output below.
i.e. eslint --ext ".ts,.js" src --debug
-->

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `4.25.0` |
| `TypeScript`                | `4.2.4` |
| `ESLint`                    | `7.27.0` |
| `node`                      | `12.22.1` |

**Comment by bradzacher** [comment]
could you please provide an isolated repro for this?
there shouldn't be any way that this happened.

**Comment by 6XGate** [comment]
I'll try this evening since this is not my code in actuality.

**Comment by 6XGate** [comment]
Okay, after digging into this, I discovered that Standard TS's config repository has somehow locked the `@typescript-eslint/eslint-plugin@4.0.1`.  Leaving the `no-extra-parens` is broken.  I'll see if I can get them to push their dependencies past `4.15.0`

**Comment by bradzacher** [comment]
That's pretty terrible! There have been many awesome features released since then (like full support for `no-unused-vars`), so it really sucks that they're pinned on a version that's like 6 months old.
